r-package/grf/R/average_treatment_effect.R

```diff
@@ -294,3 +295,39 @@ average_treatment_effect = function(forest,
   tau.se <- sqrt(sigma2.hat)
   return(c(estimate=tau.avg, std.err=tau.se))
 }
+
+
+#' Estimate average outcomes conditional on X and W given a trained causal forest
+#'
+#' @param forest The trained forest.
+#' @param subset Specifies subset of the training examples over which we
+#'               estimate the ATE. WARNING: For valid statistical performance,
+#'               the subset should be defined only using features Xi, not using
+#'               the treatment Wi or the outcome Yi.
+#' @return A list containing Y.hat.0 and Y.hat.1, the estimated W=0 and W=1 averages
+#' @export
+average_outcomes_X_W <- function(forest, subset=NULL){
```

Comment:
I didn't really understand what this name was supposed to mean. Perhaps we could rename this something like `estimate_counterfactual_outcomes`?